packages/contracts/test/transactionManager.spec.ts

```diff
@@ -20,10 +23,8 @@ import { getOnchainBalance } from "./utils";
 
 const { AddressZero } = constants;
 
-type VariantTransactionData = {
-  amount: string;
-  blockNumber: BigNumberish;
-  expiry: string;
+const advanceBlockTime = async (desiredTimestamp: number) => {
```

Comment:
These are things (including the `getWallets()` call) that may be especially difficult to replicate if we aren't using a local hardhat instance. Ideally, all of these tests would be structured such that we can easily point these contract tests at any "evm compatible" chain to make sure these things work.

As a sidenote, we'll want to be *especially* careful around the assumptions were making with `SafeMath` when pointing at various rollup flavors

Reply:
Also (i can't highlight the right lines, sorry) we need to make sure we are using *real* deployments (i.e. including the multisender contracts) for these tests and test fixtures

Reply:
To avoid confusion, we should also make sure we add a set amount of liquidity to *all* the test wallets. We'll want to structure the utils now so they fund the addresses to a given level *only* if needed (i.e. if I want 5, and have 3, I only send an additional 2. If I want 5 and have 7, I don't send anything). We did some of this in the vector integration tests

Reply:
Instead of pivoting out test setup from hardat to something compatible to run with test network. We can use fork setup to test contracts out, https://hardhat.org/guides/mainnet-forking.html.

This way we can test with any evm compatible chains.

Reply:
we are actually adding set amount of liquidity to router wallet in beforeEach